repos/system_upgrade/el7toel8/actors/checkmemory/libraries/library.py

```diff
@@ -0,0 +1,44 @@
+from leapp.models import MemoryInfo
+from leapp.libraries.common.config import architecture
+from leapp.exceptions import StopActorExecutionError
+from leapp.libraries.stdlib import api
+from leapp.reporting import create_report
+from leapp import reporting
+
+min_req_memory = {
+    architecture.ARCH_X86_64: 1536,  # 1.5G
+    architecture.ARCH_ARM64: 2048,  # 2Gb
+    architecture.ARCH_PPC64LE: 2048,  # 2Gb
+    architecture.ARCH_S390X: 1024  # 1Gb
+}
+
+
+def _check_memory(mem_info):
+    msg = {}
+
+    for arch, min_req in iter(min_req_memory.items()):
+        if architecture.matches_architecture(arch):
+            is_ok = mem_info.mem_total >= min_req
+            msg = {} if is_ok else {'memory': [mem_info.mem_total, min_req]}
+
+    return msg
+
+
+def process():
+    memoryinfo = next(api.consume(MemoryInfo), None)
+    if memoryinfo is None:
+        raise StopActorExecutionError(message=("Missing information about Memory."))
+
+    minimum_req_error = _check_memory(memoryinfo)
+
+    if minimum_req_error:
+        title = 'Minimum memory requirements for RHEL 8 are not met'
+        summary = 'Memory detected: {} KiB, required: {} KiB'.format(minimum_req_error['memory'][0],
```

Comment:
Maybe change the format of _check_memory output then? A dict {'minimal_req': X, 'detected': Y} is more obvious to use with format then {'memory': [X, Y]}